### ISSUE TITLE
How to enforce TwinsDiff#step2_materialize method to return a pair of JobTimestamp object created by user code?

### ISSUE THREAD

**Issue by kazurayam** [kazurayam/inspectus#22]
v0.5.1, TwinsDiff class 
https://github.com/kazurayam/inspectus/blob/0.5.1/core/src/main/java/com/kazurayam/inspectus/core/internal/TwinsDiff.java
has the following code:

```
    public Intermediates step3_reduceTwins(Parameters parameters,
                                           Intermediates intermediates)
            throws InspectusException {
        ...
        if (!intermediates.containsJobTimestampLeft()) {
            throw new InspectusException(Intermediates.KEY_jobTimestampLeft + " is not given in the intermediates object");
        }
        JobTimestamp jobTimestampLeft = intermediates.getJobTimestampLeft();

        if (!intermediates.containsJobTimestampRight()) {
            throw new InspectusException(Intermediates.KEY_jobTimestampRight + " is not given in the parameters");
        }
        JobTimestamp jobTimestampRight = intermediates.getJobTimestampRight();
```

These `if` statements require the `step2_materialize` method to return a `Intermediates` object to include the following 2 properites:

1. `jobTimestampLeft`
2. `jobTimestampRight`

The values of these 2 JobTimestamp object are dynamically determined by the TwinsDiff#step2_materialize method; it is not specified by man as runtime parameters.

How to tell programmers that they must populate these 2 properties? How to enforce them?

**Comment by kazurayam** [comment]
I have got an idea.

1. I will add `TwinsDiffIntermediates` class that requires the 4 properties to construct.
2. TwinsDiff#step2_materialize() should return an instance of `TwinsDiffIntermediates` object
3. TwinsDiff#step3_reduce() will accept the `TwinsDiffIntermediate` object and consume it.

**Comment by kazurayam** [comment]
I will create a new class `ParametersForTwinsDiff` which extends the `Parameters` class.
The constructor of `ParametersForTwinsDiff` will require `profileLeft` and `profileRight` as mandatory parameter. This will enforce users to specify the value of `profileLeft` and `profileRight`.

**Comment by kazurayam** [comment]
resolved at  0.5.4
